src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailableLocationsService.kt

```diff
@@ -107,15 +111,88 @@ class AvailableLocationsService(
 
   private fun getDecoratedLocationsAt(prisonCode: String) = locationsService.getDecoratedVideoLocations(prisonCode = prisonCode, enabledOnly = true)
 
-  private fun Location.allowsByAnyRuleOrSchedule(request: AvailableLocationsRequest): Boolean {
+  private fun Location.allowsByAnyRuleOrSchedule(request: AvailableLocationsRequest, time: LocalTime): Boolean {
     if (extraAttributes != null) {
       return when (request.bookingType!!) {
-        // TODO lookup attribute and court/probation team then see if applicable
-        BookingType.COURT -> true
-        BookingType.PROBATION -> true
+        BookingType.COURT -> locationAttributesService.isLocationAvailableFor(
+          LocationAvailableRequest.court(
+            extraAttributes.attributeId,
+            request.courtCode!!,
+            request.date!!.atTime(time),
+          ),
+        )
+        BookingType.PROBATION -> locationAttributesService.isLocationAvailableFor(
+          LocationAvailableRequest.probation(
+            extraAttributes.attributeId,
+            request.probationTeamCode!!,
+            request.date!!.atTime(time),
+          ),
+        )
       }
     }
 
     return true
   }
 }
+
+@Service
+@Transactional(readOnly = true)
```

Comment:
The intent on a later PR is to move all of the location related services into a sub-package e.g. locations.  As there are now a few locations related services.